react/components/buy-button/index.js

```diff
@@ -42,36 +66,38 @@ export class BuyButton extends Component {
           },
         ],
       },
-      refetchQueries: [{ query: orderFormQuery }],
-    }).then(res => {
-      const { items } = res.data.addItem
-      if (find(items, { id: skuId })) {
-        document.dispatchEvent(new Event('item:add'))
-        this.setState({ isLoading: !this.state.isLoading })
-      }
-    }, (err) => {
-      if (err) {
-        document.dispatchEvent(new Event('item:fail'))
-        this.setState({ isLoading: !this.state.isLoading })
+    }).then(
+      res => {
+        const { items } = res.data.addItem
+
+        if (find(items, { id: skuId })) {
+          this.toastMessage(true)
+        } else {
+          this.toastMessage(false)
+        }
+
+        this.setState({ isLoading: false })
+      },
+      () => {
+        this.toastMessage(false)
+        this.setState({ isLoading: false })
```

Comment:
Would be better also passtrough the `err` of the callback.

Reply:
👍